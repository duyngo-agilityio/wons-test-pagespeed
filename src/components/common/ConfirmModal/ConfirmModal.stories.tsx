import type { Meta, StoryObj } from '@storybook/react';

// components
import { ConfirmModal } from '@/components';

const meta: Meta<typeof ConfirmModal> = {
  title: 'Components/Common/ModalConfirm',
  component: ConfirmModal,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
  argTypes: {
    title: {
      control: {
        type: 'text',
      },
    },
    content: {
      control: {
        type: 'text',
      },
    },
    onConfirm: {
      action: 'onConfirm',
    },
    onCancel: {
      action: 'onCancel',
    },
    isOpen: {
      control: {
        type: 'boolean',
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof ConfirmModal>;

export const Default: Story = {
  args: {
    title: 'Delete Item',
    content: 'Are you sure you want to delete this item?',
    onConfirm: () => {},
    onCancel: () => {},
    isOpen: true,
  },
};
