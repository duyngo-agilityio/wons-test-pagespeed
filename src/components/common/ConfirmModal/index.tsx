import {
  Modal as NextUIModal,
  ModalContent,
  ModalHeader,
  ModalBody,
  ModalFooter,
} from '@nextui-org/react';

// components
import { Button } from '@/components';

interface ConfirmModalProps {
  title: string;
  content: string;
  isOpen: boolean;
  onConfirm?: () => void;
  onCancel?: () => void;
}

const ConfirmModal = ({
  title,
  isOpen,
  content,
  onConfirm,
  onCancel,
}: ConfirmModalProps) => {
  const handleSubmit = (onClose: () => void) => {
    if (onConfirm) {
      onConfirm();
    }
    onClose();
  };

  const handleCancel = (onClose: () => void) => {
    if (onCancel) {
      onCancel();
    }
    onClose();
  };

  return (
    <NextUIModal
      backdrop="opaque"
      isOpen={isOpen}
      onOpenChange={onCancel}
      radius="lg"
      classNames={{
        body: 'py-6',
        backdrop: 'bg-[#292f46]/50 backdrop-opacity-40',
        base: 'border-[#292f46] bg-white text-black dark:bg-gray-400 dark:text-white',
        closeButton: 'hover:bg-white/5 active:bg-white/10',
      }}
    >
      <ModalContent>
        {(onClose) => (
          <>
            <ModalHeader className="flex justify-center !text-xxl font-bold text-blue-800/70 dark:text-white/70">
              {title}
            </ModalHeader>

            <ModalBody className="pt-4 text-center !text-xl text-blue-800/70 dark:text-white/70">
              <p className="text-lg">{content}</p>
            </ModalBody>

            <ModalFooter className="flex justify-center gap-24">
              <Button
                className="!bg-blue-500 dark:!bg-blue-500 text-white/70 hover:!bg-blue-200 dark:hover:!bg-blue-200 px-6 py-2 !rounded-10 shadow-md"
                onPress={() => handleSubmit(onClose)}
              >
                Submit
              </Button>

              <Button
                className="!bg-white dark:!bg-white !text-blue-800/70 dark:text-white/70 border border-gray-300 hover:bg-gray-100 dark:hover:!bg-gray-300 px-6 py-2 !rounded-10"
                onPress={() => handleCancel(onClose)}
              >
                Cancel
              </Button>
            </ModalFooter>
          </>
        )}
      </ModalContent>
    </NextUIModal>
  );
};

export default ConfirmModal;
