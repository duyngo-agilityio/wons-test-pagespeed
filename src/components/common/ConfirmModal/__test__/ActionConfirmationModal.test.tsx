import ConfirmModal from '../index';

describe('ModalConfirm Component', () => {
  it('should match snapshot when open', () => {
    const mockOnConfirm = jest.fn();
    const mockOnCancel = jest.fn();

    const { container } = testLibJestUtils.render(
      <ConfirmModal
        title="Confirm Action"
        content="Are you sure you want to proceed?"
        isOpen={true}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
