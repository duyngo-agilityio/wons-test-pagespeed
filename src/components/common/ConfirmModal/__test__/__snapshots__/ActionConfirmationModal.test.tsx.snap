// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalConfirm Component should match snapshot when open 1`] = `
<div
  aria-hidden="true"
>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
     
    <div
      class="fixed top-4 right-4"
    />
  </div>
</div>
`;
